apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.html.ts

```diff
@@ -0,0 +1,156 @@
+              ${resLocals.assessment_set.name} ${resLocals.assessment.number}: Access
+            </div>
+
+            <div class="table-responsive">
+              <table class="table table-sm table-hover">
+                <thead>
+                  <tr>
+                    <th>Mode</th>
+                    <th>UIDs</th>
+                    <th>Start date</th>
+                    <th>End date</th>
+                    <th>Active</th>
+                    <th>Credit</th>
+                    <th>Time limit</th>
+                    <th>Password</th>
+                    <th>CBTF exam</th>
+                    <th>PrairieTest</th>
+                  </tr>
+                </thead>
+                <tbody>
+                  ${accessRules.map((access_rule) => {
+                    // Only users with permission to view student data are allowed
+                    // to see the list of uids associated with an access rule. Note,
+                    // however, that any user with permission to view course code
+                    // (or with access to the course git repository) will be able to
+                    // see the list of uids, because these access rules are defined
+                    // in course code. This should be changed in future, to protect
+                    // student data. See https://github.com/PrairieLearn/PrairieLearn/issues/3342
+                    return html`
+                      <tr>
+                        <td>${access_rule.mode}</td>
+                        <td>
+                          ${access_rule.uids === '—' ||
+                          resLocals.authz_data.has_course_instance_permission_view
+                            ? access_rule.uids
+                            : html`
+                                <a
+                                  role="button"
+                                  class="btn btn-xs btn-warning"
+                                  tabindex="0"
+                                  data-toggle="popover"
+                                  data-trigger="focus"
+                                  data-container="body"
+                                  data-placement="auto"
+                                  title="Hidden UIDs"
+                                  data-content="This access rule is specific to individual students. You need permission to view student data in order to see which ones."
+                                >
+                                  Hidden
+                                </a>
+                              `}
+                        </td>
+                        <td>${access_rule.start_date}</td>
+                        <td>${access_rule.end_date}</td>
+                        <td>${access_rule.active}</td>
+                        <td>${access_rule.credit}</td>
+                        <td>${access_rule.time_limit}</td>
+                        <td>${access_rule.password}</td>
+                        <td>${access_rule.exam}</td>
```

Comment:
@mwest1066 the SQL query itself is directly producing HTML like `<a href="https://cbtf.engr.illinois.edu/sched/course/...`. I'm wondering if we can safely drop this entire "CBTF exam" column now? My only reservation is the fact that people seem to think China is using something from the old scheduler, though maybe I'm misunderstanding that.

Myles, if we can safely remove this, you can clean up the query, the place the query is called, and the `syncExamIdAccessRules` config option, which is only used on this page.

Reply:
@mylesw27 discussed with Matt. This is dead code and should be removed. In fact, because this column wasn't in the existing `AssessmentAccessRulesSchema` schema, the value for `exam` in all these rows hasn't been anything but `undefined` for some time now!

Reply:
Removed in [5c7aa94](https://github.com/PrairieLearn/PrairieLearn/pull/9420/commits/5c7aa941195d3c30ddb12843a28f0e6bf548ec83).